.github/workflows/klipper3d-deploy.yaml

```diff
@@ -8,16 +9,30 @@ on:
     paths:
       - docs/**
       - .github/workflows/klipper3d-deploy.yaml
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Documentation page deployment source branch (eg: "pr-docs")'
+        required: true
+        default: master
+
+# Required for the deployment to be able to upload to the "gh-pages" branch
+# (since the repo settings for default actions permission defaults to read only)
+permissions:
+  contents: write
```

Comment:
I'm leery of making a permission change like this.  My fear is that one of the many build scripts could alter the master branch (which would then propagate to thousands of printers to within a few days).  Not sure the trade-off on this commit is worthwhile - might be better to just add documentation on how one can build the documentation (I use a docker instance, for example).

Reply:
Ok but your current setup is less secure.
Without write permissions all current workflows in this repo should fail. Therefore I guess at the repo or organization level the default github workflow permission setting is set to "read & write".
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token

![Screenshot_20240217-090544.png](https://github.com/Klipper3d/klipper/assets/21227087/70656568-e80f-4df0-a924-ca98ca693f63)

If a threaded harmfull gets merged now it has write permissions to everything.

If your current default token permissions would be set to read only [the pages deployment action](https://github.com/Klipper3d/klipper/blob/72b301a2859c3f7ed26d802dd52fc495eef6c353/.github/workflows/klipper3d-deploy.yaml#L31) would error like this:
```
Error: The deploy step encountered an error: The process '/usr/bin/git' failed with exit code 128 ❌
```